src/peft/tuners/adalora/config.py

```diff
@@ -68,10 +68,13 @@ def __post_init__(self):
         if isinstance(self.target_modules, str) and self.layers_to_transform is not None:
             raise ValueError("`layers_to_transform` cannot be used when `target_modules` is a str.")
 
-        # if target_modules is a regex expression, then layers_pattern should be None
-        if isinstance(self.target_modules, str) and self.layers_pattern is not None:
-            raise ValueError("`layers_pattern` cannot be used when `target_modules` is a str.")
+        # check for layers_to_transform and layers_pattern
+        if (self.layers_pattern is not None) and (self.layers_to_transform is None):
```

Comment:
I think these checks should be adjusted like so:
```suggestion
        if self.layers_pattern and not self.layers_to_transform:
```

Why? 

1. If users pass `layers_pattern=[], layers_to_transform=None`, we should not raise.
2. If users pass `layers_pattern=["foo"], layers_to_transform=[]`, we should raise

Right now, we don't do that because of the strict `None` check.